src/qt/treectrl.cpp

```diff
@@ -8,11 +8,438 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+#pragma hdrstop
+#endif
+
 #include "wx/treectrl.h"
+#include "wx/imaglist.h"
+
+#include "wx/qt/private/winevent.h"
+#include "wx/qt/private/treeitemfactory.h"
 
 #include <QtWidgets/QTreeWidget>
+#include <QtWidgets/QHeaderView>
+#include <QtGui/QPainter>
+
+namespace
+{
+struct TreeItemDataQt
+{
+    TreeItemDataQt() : data(NULL)
+    {
+    }
+
+    TreeItemDataQt(wxTreeItemData* data) : data(data)
+    {
+        if ( !registered )
+        {
+            qRegisterMetaTypeStreamOperators<TreeItemDataQt>("TreeItemDataQt");
+            registered = true;
+        }
+    }
+
+    wxTreeItemData *data;
+
+    private:
+    static bool registered;
+};
+
+bool TreeItemDataQt::registered = false;
+Q_DECLARE_METATYPE(TreeItemDataQt)
+
+QDataStream &operator<<(QDataStream &out, const TreeItemDataQt &WXUNUSED(obj))
+{
+    return out;
+}
+QDataStream &operator>>(QDataStream &in, TreeItemDataQt &WXUNUSED(obj))
+{
+    return in;
+}
+
+QTreeWidgetItem *wxQtConvertTreeItem(const wxTreeItemId &item)
+{
+    return static_cast<QTreeWidgetItem*>(item.GetID());
+}
+
+wxTreeItemId wxQtConvertTreeItem(QTreeWidgetItem *item)
+{
+    return wxTreeItemId(item);
+}
 
-wxTreeCtrl::wxTreeCtrl()
+size_t CountChildren(QTreeWidgetItem *item)
+{
+    const int currentCount = item->childCount();
+    size_t totalCount = currentCount;
+
+    for ( int i = 0; i < currentCount; ++i )
+    {
+        totalCount += CountChildren(item->child(i));
+    }
+
+    return totalCount;
+}
+class ImageState
+{
+public:
+    ImageState() : m_state(wxTREE_ITEMSTATE_NONE)
+    {
+        for ( int i = wxTreeItemIcon_Normal; i < wxTreeItemIcon_Max; ++i )
+        {
+            m_imageStates[i] = -1;
```

Comment:
Please prefer using symbolic constants:
```suggestion
            m_imageStates[i] = wxWithImages::NO_IMAGE;
```